### PR TITLE
feat: Optimize E2E test actions

### PR DESCRIPTION
**Pull request by 0x6e616d**
Based on the circle config `.circleci/config.yml`, I optimized the e2e tests to many smaller actions to robust the speed and reduce the cost of running the tests.

This PR will be tested again after the [PR](https://github.com/tokamak-network/tokamak-thanos/pull/194) is merged(to test the op-bindings folder)

### CODE REVIEW

File: op-ufm/go.mod [hunk 1]
```diff
@@ -24,8 +24,8 @@ require (
 	github.com/Microsoft/go-winio v0.6.1 // indirect
```
Comment (rlgns98kr) [inline op-ufm/go.mod:24]: Can you explain more about why there is some changes in our module?

Comment (0x6e616d) [reply op-ufm/go.mod:24]: Maybe these changes are from my IDE, it indexed the go.mod automatically. Let me revert go.mod and go.sum files

File: .github/workflows/go-test.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
+name: Go E2E tests
+
+on:
+  push:
+    branches:
+      - feat/OR-1618-seperate-e2e-test
+
+jobs:
+  go-test:
+    runs-on: ubuntu-latest-l
```
Comment (rlgns98kr) [inline .github/workflows/go-test.yml:10]: Thank you very much, Nam!
Can you use the `ubuntu`?

**Comment by nguyenzung** [comment]
Hi @0x6e616d 
Do you think we should rebuild and setup e2e-test? Correct me If I am wrong, the action uses cache for fast setting?

**Comment by 0x6e616d** [comment]
> Hi @0x6e616d Do you think we should rebuild and setup e2e-test? Correct me If I am wrong, the action uses cache for fast setting?

The purpose is to split e2e-tests into smaller tests and run them in parallel to reduce the ci time. I use all test actions in `config.yaml` from circle ci and convert them to GitHub actions

**Comment by nguyenzung** [comment]
> > Hi @0x6e616d Do you think we should rebuild and setup e2e-test? Correct me If I am wrong, the action uses cache for fast setting?
> 
> The purpose is to split e2e-tests into smaller tests and run them in parallel to reduce the ci time. I use all test actions in `config.yaml` from circle ci and convert them to GitHub actions



> > Hi @0x6e616d Do you think we should rebuild and setup e2e-test? Correct me If I am wrong, the action uses cache for fast setting?
> 
> The purpose is to split e2e-tests into smaller tests and run them in parallel to reduce the ci time. I use all test actions in `config.yaml` from circle ci and convert them to GitHub actions

Thank you, Nam!

I mean it seems the action is using a cache for fast setting, because I don't see any build steps. I don't know if the test can run correctly or not if the cache is not up to date.

**Comment by 0x6e616d** [comment]
> > > Hi @0x6e616d Do you think we should rebuild and setup e2e-test? Correct me If I am wrong, the action uses cache for fast setting?
> > 
> > 
> > The purpose is to split e2e-tests into smaller tests and run them in parallel to reduce the ci time. I use all test actions in `config.yaml` from circle ci and convert them to GitHub actions
> 
> > > Hi @0x6e616d Do you think we should rebuild and setup e2e-test? Correct me If I am wrong, the action uses cache for fast setting?
> > 
> > 
> > The purpose is to split e2e-tests into smaller tests and run them in parallel to reduce the ci time. I use all test actions in `config.yaml` from circle ci and convert them to GitHub actions
> 
> Thank you, Nam!
> 
> I mean it seems the action is using a cache for fast setting, because I don't see any build steps. I don't know if the test can run correctly or not if the cache is not up to date.
> 
> And for optimization purpose, we can define the number of parallel threads for e2e-test. I am not sure if it can help reduce time, but I think we can check the log easier than the current appoarch.

Oh no, the old version includes the build step and all the tests in the folder `op-e2e`. But my version includes the tests with each sub folder `op-batcher`, `op-challenger`...This is quite a different purpose

**Review by rlgns98kr — APPROVED** [review]
Thank you very much, Nam! 🙇🏼‍♂️

**Review by rlgns98kr — APPROVED** [review]
Thank you very much!

**Pull request merged**